src/optimization/Optimizer.h

```diff
@@ -101,18 +102,23 @@ namespace vc4c
 		extern const OptimizationPass EXTEND_BRANCHES;
 		//adds the start- and stop-segments to the beginning and end of the kernel
 		extern const OptimizationPass ADD_START_STOP_SEGMENT;
+		//remove unneccesary "and" and "or"
+		extern const OptimizationPass REMOVE_REDUNDANT_BITOP;
 
+		extern const OptimizationPass PROPAGATE_VAR;
 		/*
 		 * The default optimization passes consist of all passes listed above.
 		 * NOTE: Some of the passes are REQUIRED and the compilation will fail, if they are removed.
 		 * Other passes are not technically required, but e.g. make register-allocation a lot easier, thus improving the chance of successful register allocation greatly.
 		 */
-		extern const std::set<OptimizationPass> DEFAULT_PASSES;
+		extern const std::vector<OptimizationPass> DEFAULT_PASSES;
+
+	    extern const OptimizationPass TRANSLATE_TO_MOVE;
 
 		class Optimizer
 		{
 		public:
-			Optimizer(const Configuration& config = { }, const std::set<OptimizationPass>& passes = DEFAULT_PASSES);
+			Optimizer(const Configuration& config = { }, const std::vector<OptimizationPass>& passes = DEFAULT_PASSES);
```

Comment:
Is there a reason, you change the set to vector?
Currently the last parameter of the `OptimizationPass` constructor is used to sort the optimizations, with the options to add/remove optimizations in the correct position.
Since there is no real use-case for adding/removing optimizations, we could use a vector here. In this case, we should remove the now superfluous index as well as the add/remove functions.

Reply:
I change it to vector because the same optimizations should work in multiple times.
Come to think of it, defining the same optimization in the same name like `REMOVE_REDUNDANT_MOVES_1` and `REMOVE_REDUNDANT_MOVES_2`.

I will revert this change.